### PR TITLE
Feature/kyv 393 right of purchase

### PR DESCRIPTION
**Pull request by severikupari1**

### CODE REVIEW

File: common/src/main/java/fi/hel/verkkokauppa/common/rest/RestServiceClient.java [hunk 1]
```diff
@@ -29,13 +29,21 @@
     public JSONObject makeGetCall(String url) {
         WebClient client = getClient();
         JSONObject response = queryJsonService(client, url);
-        return response;
+        if (response == null) {
```
Comment (severikupari1) [inline common/src/main/java/fi/hel/verkkokauppa/common/rest/RestServiceClient.java:32]: These fixes NullPointerExceptions

**Review by hleisma — CHANGES_REQUESTED** [review]
Avoid chaining throws JsonProcessingException, that just makes your problem somebody elses problem. We have agreed that REST endpoints only return structured error messages, not ok to let an exception through.

File: orderapi/src/main/java/fi/hel/verkkokauppa/order/api/SubscriptionAdminController.java [hunk 2]
```diff
@@ -85,7 +86,7 @@
     }
 
     @PostMapping(value = "/subscription-admin/renewal-requested-event", produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<Void> subscriptionRenewalRequestedEventCallback(@RequestBody SubscriptionMessage message) {
+    public ResponseEntity<Void> subscriptionRenewalRequestedEventCallback(@RequestBody SubscriptionMessage message) throws JsonProcessingException {
```
[the remark this answers is not in the document]
Comment (severikupari1) [reply orderapi/src/main/java/fi/hel/verkkokauppa/order/api/SubscriptionAdminController.java:89]: I will fix these error throwings. Good to know to catch all errors and just log and change http status. Good points, thanks for review!

[the remark this answers is not in the document]
Comment (severikupari1) [reply orderapi/src/main/java/fi/hel/verkkokauppa/order/api/SubscriptionAdminController.java]: Now this is fixed.

**Pull request merged**